File: src/main/java/com/wgzhao/addax/admin/dto/EtlTask.java
package com.wgzhao.addax.admin.dto;

import java.util.Map;

/**
 * 采集任务数据结构
 */
public class EtlTask {
    private final String taskId;
    private final String taskType;
    private final Map<String, Object> taskData;
    private final long createTime;

    public EtlTask(String taskId, String taskType, Map<String, Object> taskData) {
        this.taskId = taskId;
        this.taskType = taskType;
        this.taskData = taskData;
        this.createTime = System.currentTimeMillis();
    }

    // getters
    public String getTaskId() { return taskId; }
    public String getTaskType() { return taskType; }
    public Map<String, Object> getTaskData() { return taskData; }
    public long getCreateTime() { return createTime; }

    @Override
    public String toString() {
        return String.format("EtlTask{taskId='%s', taskType='%s', createTime=%d}",
                           taskId, taskType, createTime);
    }
}

File: src/main/kotlin/com/wgzhao/addax/admin/dto/EtlTask.kt
package com.wgzhao.addax.admin.dto

/**
 * 采集任务数据结构
 */
class EtlTask(// getters
    val taskId: String?, val taskType: String?, val taskData: MutableMap<String?, Any?>?
) {
    val createTime: Long

    init {
        this.createTime = System.currentTimeMillis()
    }

    override fun toString(): String {
        return String.format(
            "EtlTask{taskId='%s', taskType='%s', createTime=%d}",
            taskId, taskType, createTime
        )
    }
}
